src/Symfony/Bundle/TwigBundle/Loader/NativeFilesystemLoader.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\TwigBundle\Loader;
+
+use Twig\Error\LoaderError;
+use Twig\Loader\FilesystemLoader;
+
+class NativeFilesystemLoader extends FilesystemLoader
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function findTemplate($template, $throw = true)
+    {
+        try {
+            return parent::findTemplate($template, $throw);
+        } catch (LoaderError $e) {
+            if ($e instanceof \Twig_Error_Loader && preg_match('/^([^:]*?)(?:Bundle)?:([^:]*+):(.+\.[^\.]+\.[^\.]+)$/', $template, $m) && 0 !== strpos($template, '@')) {
```

Comment:
You should switch the regex match in the `strpos`. Checking the first char is faster.

Btw, you could optimize it even more:

```php
if ('' !== $template && '@' !== $template[0] && preg_match(...)) {
```

Using strpos to check only the first char is not the fastest way.